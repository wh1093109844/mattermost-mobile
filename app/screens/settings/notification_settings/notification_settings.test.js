// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';

import Preferences from 'mattermost-redux/constants/preferences';

import {shallowWithIntl} from 'test/intl-test-helper';

import NotificationSettings from './notification_settings.js';

describe('NotificationSettings', () => {
    const baseProps = {
        actions: {
            updateMe: jest.fn(),
        },
        componentId: 'component-id',
        currentUser: {id: 'current_user_id'},
        theme: Preferences.THEMES.default,
        updateMeRequest: {},
        currentUserStatus: 'status',
        enableAutoResponder: false,
        isLandscape: false,
    };

    test('should match snapshot', () => {
        const wrapper = shallowWithIntl(
            <NotificationSettings {...baseProps}/>
        );

        expect(wrapper.instance()).toMatchSnapshot();
    });

    test('should include previous notification props when saving new ones', () => {
        baseProps.currentUser.notify_props = {previous: 'previous'};
        const wrapper = shallowWithIntl(
            <NotificationSettings {...baseProps}/>
        );

        const instance = wrapper.instance();
        const newProps = {new: 'new'};
        instance.saveNotificationProps(newProps);
        expect(baseProps.actions.updateMe).toHaveBeenCalledWith({
            notify_props: {
                ...baseProps.currentUser.notify_props,
                ...newProps,
            },
        });
    });
});
