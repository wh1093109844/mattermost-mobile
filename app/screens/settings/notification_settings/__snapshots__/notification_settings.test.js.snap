// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationSettings should match snapshot 1`] = `
NotificationSettings {
  "context": Object {
    "intl": Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": Object {},
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    },
  },
  "goToNotificationSettingsAutoResponder": [Function],
  "goToNotificationSettingsEmail": [Function],
  "goToNotificationSettingsMentions": [Function],
  "goToNotificationSettingsMobile": [Function],
  "handlePress": [Function],
  "props": Object {
    "actions": Object {
      "updateMe": [MockFunction],
    },
    "componentId": "component-id",
    "currentUser": Object {
      "id": "current_user_id",
    },
    "currentUserStatus": "status",
    "enableAutoResponder": false,
    "intl": Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": Object {},
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    },
    "isLandscape": false,
    "theme": Object {
      "awayIndicator": "#ffbc42",
      "buttonBg": "#166de0",
      "buttonColor": "#ffffff",
      "centerChannelBg": "#ffffff",
      "centerChannelColor": "#3d3c40",
      "codeTheme": "github",
      "dndIndicator": "#f74343",
      "errorTextColor": "#fd5960",
      "linkColor": "#2389d7",
      "mentionBg": "#ffffff",
      "mentionBj": "#ffffff",
      "mentionColor": "#145dbf",
      "mentionHighlightBg": "#ffe577",
      "mentionHighlightLink": "#166de0",
      "newMessageSeparator": "#ff8800",
      "onlineIndicator": "#06d6a0",
      "sidebarBg": "#145dbf",
      "sidebarHeaderBg": "#1153ab",
      "sidebarHeaderTextColor": "#ffffff",
      "sidebarText": "#ffffff",
      "sidebarTextActiveBorder": "#579eff",
      "sidebarTextActiveColor": "#ffffff",
      "sidebarTextHoverBg": "#4578bf",
      "sidebarUnreadText": "#ffffff",
      "type": "Mattermost",
    },
    "updateMeRequest": Object {},
  },
  "refs": Object {},
  "saveAutoResponder": [Function],
  "saveNotificationProps": [Function],
  "setState": [Function],
  "shouldSaveAutoResponder": [Function],
  "state": null,
  "updater": Updater {
    "_callbacks": Array [],
    "_renderer": ReactShallowRenderer {
      "_context": Object {
        "intl": Object {
          "defaultFormats": Object {},
          "defaultLocale": "en",
          "formatDate": [Function],
          "formatHTMLMessage": [Function],
          "formatMessage": [Function],
          "formatNumber": [Function],
          "formatPlural": [Function],
          "formatRelative": [Function],
          "formatTime": [Function],
          "formats": Object {},
          "formatters": Object {
            "getDateTimeFormat": [Function],
            "getMessageFormat": [Function],
            "getNumberFormat": [Function],
            "getPluralFormat": [Function],
            "getRelativeFormat": [Function],
          },
          "locale": "en",
          "messages": Object {},
          "now": [Function],
          "onError": [Function],
          "textComponent": "span",
          "timeZone": null,
        },
      },
      "_didScheduleRenderPhaseUpdate": false,
      "_dispatcher": Object {
        "readContext": [Function],
        "useCallback": [Function],
        "useContext": [Function],
        "useDebugValue": [Function],
        "useEffect": [Function],
        "useImperativeHandle": [Function],
        "useLayoutEffect": [Function],
        "useMemo": [Function],
        "useReducer": [Function],
        "useRef": [Function],
        "useResponder": [Function],
        "useState": [Function],
      },
      "_element": <NotificationSettings
        actions={
          Object {
            "updateMe": [MockFunction],
          }
        }
        componentId="component-id"
        currentUser={
          Object {
            "id": "current_user_id",
          }
        }
        currentUserStatus="status"
        enableAutoResponder={false}
        intl={
          Object {
            "defaultFormats": Object {},
            "defaultLocale": "en",
            "formatDate": [Function],
            "formatHTMLMessage": [Function],
            "formatMessage": [Function],
            "formatNumber": [Function],
            "formatPlural": [Function],
            "formatRelative": [Function],
            "formatTime": [Function],
            "formats": Object {},
            "formatters": Object {
              "getDateTimeFormat": [Function],
              "getMessageFormat": [Function],
              "getNumberFormat": [Function],
              "getPluralFormat": [Function],
              "getRelativeFormat": [Function],
            },
            "locale": "en",
            "messages": Object {},
            "now": [Function],
            "onError": [Function],
            "textComponent": "span",
            "timeZone": null,
          }
        }
        isLandscape={false}
        theme={
          Object {
            "awayIndicator": "#ffbc42",
            "buttonBg": "#166de0",
            "buttonColor": "#ffffff",
            "centerChannelBg": "#ffffff",
            "centerChannelColor": "#3d3c40",
            "codeTheme": "github",
            "dndIndicator": "#f74343",
            "errorTextColor": "#fd5960",
            "linkColor": "#2389d7",
            "mentionBg": "#ffffff",
            "mentionBj": "#ffffff",
            "mentionColor": "#145dbf",
            "mentionHighlightBg": "#ffe577",
            "mentionHighlightLink": "#166de0",
            "newMessageSeparator": "#ff8800",
            "onlineIndicator": "#06d6a0",
            "sidebarBg": "#145dbf",
            "sidebarHeaderBg": "#1153ab",
            "sidebarHeaderTextColor": "#ffffff",
            "sidebarText": "#ffffff",
            "sidebarTextActiveBorder": "#579eff",
            "sidebarTextActiveColor": "#ffffff",
            "sidebarTextHoverBg": "#4578bf",
            "sidebarUnreadText": "#ffffff",
            "type": "Mattermost",
          }
        }
        updateMeRequest={Object {}}
      />,
      "_firstWorkInProgressHook": null,
      "_forcedUpdate": false,
      "_instance": [Circular],
      "_isReRender": false,
      "_newState": null,
      "_numberOfReRenders": 0,
      "_renderPhaseUpdates": null,
      "_rendered": <View
        style={
          Object {
            "backgroundColor": "#ffffff",
            "flex": 1,
          }
        }
      >
        <Connect(StatusBar) />
        <ScrollViewMock
          alwaysBounceVertical={false}
          contentContainerStyle={
            Object {
              "backgroundColor": "rgba(61,60,64,0.06)",
              "paddingTop": 35,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "rgba(61,60,64,0.1)",
                "height": 1,
                "width": "100%",
              }
            }
          />
          <SettingsItem
            defaultMessage="Mentions and Replies"
            i18nId="mobile.notification_settings.mentions_replies"
            iconName="md-at"
            iconType="ion"
            isDestructor={false}
            isLandscape={false}
            onPress={[Function]}
            separator={true}
            showArrow={true}
            theme={
              Object {
                "awayIndicator": "#ffbc42",
                "buttonBg": "#166de0",
                "buttonColor": "#ffffff",
                "centerChannelBg": "#ffffff",
                "centerChannelColor": "#3d3c40",
                "codeTheme": "github",
                "dndIndicator": "#f74343",
                "errorTextColor": "#fd5960",
                "linkColor": "#2389d7",
                "mentionBg": "#ffffff",
                "mentionBj": "#ffffff",
                "mentionColor": "#145dbf",
                "mentionHighlightBg": "#ffe577",
                "mentionHighlightLink": "#166de0",
                "newMessageSeparator": "#ff8800",
                "onlineIndicator": "#06d6a0",
                "sidebarBg": "#145dbf",
                "sidebarHeaderBg": "#1153ab",
                "sidebarHeaderTextColor": "#ffffff",
                "sidebarText": "#ffffff",
                "sidebarTextActiveBorder": "#579eff",
                "sidebarTextActiveColor": "#ffffff",
                "sidebarTextHoverBg": "#4578bf",
                "sidebarUnreadText": "#ffffff",
                "type": "Mattermost",
              }
            }
          />
          <SettingsItem
            defaultMessage="Mobile"
            i18nId="mobile.notification_settings.mobile"
            iconName="md-phone-portrait"
            iconType="ion"
            isDestructor={false}
            isLandscape={false}
            onPress={[Function]}
            separator={true}
            showArrow={true}
            theme={
              Object {
                "awayIndicator": "#ffbc42",
                "buttonBg": "#166de0",
                "buttonColor": "#ffffff",
                "centerChannelBg": "#ffffff",
                "centerChannelColor": "#3d3c40",
                "codeTheme": "github",
                "dndIndicator": "#f74343",
                "errorTextColor": "#fd5960",
                "linkColor": "#2389d7",
                "mentionBg": "#ffffff",
                "mentionBj": "#ffffff",
                "mentionColor": "#145dbf",
                "mentionHighlightBg": "#ffe577",
                "mentionHighlightLink": "#166de0",
                "newMessageSeparator": "#ff8800",
                "onlineIndicator": "#06d6a0",
                "sidebarBg": "#145dbf",
                "sidebarHeaderBg": "#1153ab",
                "sidebarHeaderTextColor": "#ffffff",
                "sidebarText": "#ffffff",
                "sidebarTextActiveBorder": "#579eff",
                "sidebarTextActiveColor": "#ffffff",
                "sidebarTextHoverBg": "#4578bf",
                "sidebarUnreadText": "#ffffff",
                "type": "Mattermost",
              }
            }
          />
          <SettingsItem
            defaultMessage="Email"
            i18nId="mobile.notification_settings.email"
            iconName="ios-mail"
            iconType="ion"
            isDestructor={false}
            isLandscape={false}
            onPress={[Function]}
            separator={false}
            showArrow={true}
            theme={
              Object {
                "awayIndicator": "#ffbc42",
                "buttonBg": "#166de0",
                "buttonColor": "#ffffff",
                "centerChannelBg": "#ffffff",
                "centerChannelColor": "#3d3c40",
                "codeTheme": "github",
                "dndIndicator": "#f74343",
                "errorTextColor": "#fd5960",
                "linkColor": "#2389d7",
                "mentionBg": "#ffffff",
                "mentionBj": "#ffffff",
                "mentionColor": "#145dbf",
                "mentionHighlightBg": "#ffe577",
                "mentionHighlightLink": "#166de0",
                "newMessageSeparator": "#ff8800",
                "onlineIndicator": "#06d6a0",
                "sidebarBg": "#145dbf",
                "sidebarHeaderBg": "#1153ab",
                "sidebarHeaderTextColor": "#ffffff",
                "sidebarText": "#ffffff",
                "sidebarTextActiveBorder": "#579eff",
                "sidebarTextActiveColor": "#ffffff",
                "sidebarTextHoverBg": "#4578bf",
                "sidebarUnreadText": "#ffffff",
                "type": "Mattermost",
              }
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "rgba(61,60,64,0.1)",
                "height": 1,
                "width": "100%",
              }
            }
          />
        </ScrollViewMock>
      </View>,
      "_rendering": false,
      "_updater": [Circular],
      "_workInProgressHook": null,
    },
  },
}
`;
